src/GuiBase/Viewer/Gizmo/Gizmo.cpp

```diff
@@ -68,20 +89,22 @@ bool Gizmo::findPointOnPlane( const Engine::Camera& cam,
     return hasHit;
 }
 
-void Gizmo::addRenderObject( Engine::RenderObject* ro, const std::shared_ptr<Engine::Mesh>& mesh ) {
-    m_meshes.push_back( mesh );
-    m_renderObjects.push_back( m_comp->addRenderObject( ro ) );
+void Gizmo::addRenderObject( Engine::RenderObject* ro ) {
+    m_comp->addRenderObject( ro );
+    m_ros.push_back( ro );
 }
 
-Engine::RenderTechnique* Gizmo::makeRenderTechnique( const std::string& mtlName,
-                                                     bool rtPerVertexColor ) {
+void Gizmo::changeMat( uint ro, uint mat ) {
+    m_ros[ro]->setMaterial( s_material[mat] );
+    m_ros[ro]->getRenderTechnique()->setParametersProvider( s_material[mat] );
```

Comment:
This is the more problematic modification. This restrict the use of gizmos to only one renderer and might prevent an external renderer to define its own parameter providers. This call sets the same parameter provider for all the registered passes. Even if gizmos (and other Ui or Debug components) might be drawn in a forward-rendering way. It should be more efficient and less restrictive to implement the solution I suggest in the overall comment.